ArtOfIllusion/src/artofillusion/PluginRegistry.java

```diff
@@ -36,6 +35,35 @@ Changes copyright (C) 2018-2022 by Maksim Khramov
   private static final HashMap<String, ExportInfo> exports = new HashMap<String, ExportInfo>();
   private static final HashMap<String, Object> classMap = new HashMap<String, Object>();
 
+  static {
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+        List<PluginRegistry.PluginThrowable> errors = PluginRegistry.notifyPlugins(Plugin.class, Plugin.APPLICATION_STOPPING);
+        errors.forEach(error -> System.out.println("Plugin " + error.getSource().getClass().getSimpleName() + " throws error: " + error.getCause().getMessage()) );
+    }, "Plugin shutdown thread"));
+  }
+  
+  
+  public static <T> List<PluginThrowable> notifyPlugins(Class<T> category, int message, Object... args)
```

Comment:
I'm still unsure if we should be returning an error list here at all. I'd prefer the calling code to not have to worry about this at all.

If we *do* need this, why not return a `Map <Plugin, Throwable>` instead of this custom type?